src/main/native/jni_utils.cc

```diff
@@ -21,38 +20,130 @@ JNIEnv *getJNIEnv() {
   return pEnv;
 }
 
+// Class fully qualified names
+const char *CLS_NODE = "org/bblfsh/client/v2/Node";
+const char *CLS_CTX = "org/bblfsh/client/v2/Context";
+const char *CLS_OBJ = "java/lang/Object";
+const char *CLS_RE = "java/lang/RuntimeException";
+
+const char *CLS_JNODE = "org/bblfsh/client/v2/JNode";
+const char *CLS_JNULL = "org/bblfsh/client/v2/JNull";
+const char *CLS_JSTR = "org/bblfsh/client/v2/JString";
+const char *CLS_JINT = "org/bblfsh/client/v2/JInt";
+const char *CLS_JFLT = "org/bblfsh/client/v2/JFloat";
+const char *CLS_JBOL = "org/bblfsh/client/v2/JBool";
+const char *CLS_JUINT = "org/bblfsh/client/v2/JUint";
+const char *CLS_JARR = "org/bblfsh/client/v2/JArray";
+const char *CLS_JOBJ = "org/bblfsh/client/v2/JObject";
+
+// Method signatures
+const char *METHOD_JNODE_KEY_AT = "(I)Ljava/lang/String;";
+const char *METHOD_JNODE_VALUE_AT = "(I)Lorg/bblfsh/client/v2/JNode;";
+const char *METHOD_JOBJ_ADD =
+    "(Ljava/lang/String;Lorg/bblfsh/client/v2/JNode;)Lscala/collection/"
+    "mutable/Buffer;";
+const char *METHOD_JARR_ADD =
+    "(Lorg/bblfsh/client/v2/JNode;)Lscala/collection/mutable/Buffer;";
+
+const char *METHOD_OBJ_TO_STR = "()Ljava/lang/String;";
+
+void checkJvmException(std::string msg) {
+  JNIEnv *env = getJNIEnv();
+  if (env->ExceptionCheck()) {
+    // env->ExceptionDescribe(); // prints to stdout, un-comment for debuging
```

Comment:
[optional] Maybe we should leave this in, and put it behind a flag? It can be tricky to repro issues arising from control flow across the FFI boundary, so being able to reload the binary with a debugging flag rather than recompile the whole application may help here (and we can't log in in the "normal" way otherwise at this point in the program). What do you think?

Reply:
I see what you mean, but think that I would prefer to simply remove this, if anybody feel strongly agains having this comment here.

Having a flag + reloading the native library from the JVM runtime sounds too error-prone and I do not think it will worth it. Especially, given the fact that one has to rebuild the native part with the clang debug flag anyway, in order to preserve the symbols.